codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsAuthPlugin.java

```diff
@@ -104,17 +115,63 @@ public void addConfigInterfaceFields(
             case NODE:
                 return MapUtils.of(
                     "credentialDefaultProvider", writer -> {
+                        if (!testServiceId(service, "STS")) {
+                            writer.addDependency(AwsDependency.STS_CLIENT);
+                            writer.addImport("decorateDefaultCredentialProvider", "decorateDefaultCredentialProvider",
+                                    AwsDependency.STS_CLIENT.packageName);
+                        } else {
+                            writer.addImport("decorateDefaultCredentialProvider", "decorateDefaultCredentialProvider",
+                                    "./" + ROLE_ASSUMERS_FILE);
```

Comment:
During peer programming session, we noticed that `decorateDefaultCredentialProvider` should be imported from STS_ROLE_ASSUMERS_FILE
```suggestion
                                    "./" + STS_ROLE_ASSUMERS_FILE);
```